staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -105,15 +105,24 @@ func proxyError(w http.ResponseWriter, req *http.Request, error string, code int
 }
 
 func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	httpStatus := http.StatusOK
+	serviceName := ""
+	extensionApiserverStart := time.Now()
+	ctx := req.Context()
+	defer func() {
+		recordExtensionApiserverMetrics(ctx, httpStatus, extensionApiserverStart, serviceName)
```

Comment:
Passing the apiservice as the resource name will not make correlation between apiserver_requests_duration and extension_requests_duration possible since the values of the resource labels will be different.

@benluddy had a pretty neat suggestion in https://github.com/kubernetes/kubernetes/pull/118636#discussion_r1251215059, to use group/version instead

Reply:
@dgrisonnet Group/version cannot be read in the code context, so in the latest modification I use serviceNamespace and serviceName to form the resource name. If you have time, please help to see if this is feasible.